File: HRSystem.Domain/Attributes/Base/AttributeInfo.cs
using System;
using HRSystem.Common.Errors;
using HRSystem.Global;

namespace HRSystem.Domain.Attributes.Base
{
    // ReSharper disable once ClassNeverInstantiated.Global
    public class AttributeInfo
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public AttributeType Type { get; set; }

        public bool IsActiveDirectoryAttribute => ActiveDirectoryAttributeInfoId.HasValue;

        public int? ActiveDirectoryAttributeInfoId { get; set; }

        public ActiveDirectoryAttributeInfo ActiveDirectoryAttributeInfo { get; set; }

        public AttributeInfo()
        {
        }

        public void Update(AttributeInfo attributeInfo)
        {
            Name = attributeInfo.Name;
            Type = attributeInfo.Type;
        }
    }
}

File: HRSystem.Web/ClientApp/models/AttributeInfo.ts
import {AttributeType} from "./AttributeType";
import {ActiveDirectoryAttributeInfo} from "./ActiveDirectoryAttributeInfo";

export class AttributeInfo {
    public id: number;
    public name: string;
    public type: AttributeType;
    public activeDirectoryAttributeInfo: ActiveDirectoryAttributeInfo;

    public constructor(attribute?: AttributeInfo) {
        if (attribute == null) {
            return;
        }

        this.id = attribute.id;
        this.name = attribute.name;
        this.type = attribute.type;
        if (attribute.activeDirectoryAttributeInfo != null) {
            this.activeDirectoryAttributeInfo = new ActiveDirectoryAttributeInfo(attribute.activeDirectoryAttributeInfo);
        }
    }
}